From 59a20e5e3a5461ae68d3d6b7fefde5de2480a820 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 5 May 2019 22:11:16 +0200
Subject: [PATCH] meson: fix build with opencv=enabled and opencv4. Fixes #964

Having the opencv feature enabled would lead to the opencv3 dependency
be required which failed with only opencv4 being available.

Instead don't require anything and error out at the end if the feature was enabled
but no dependency was found.
---
 gst-libs/gst/opencv/meson.build | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gst-libs/gst/opencv/meson.build b/gst-libs/gst/opencv/meson.build
index 3fd472fda..8dc2ad99b 100644
--- a/gst-libs/gst/opencv/meson.build
+++ b/gst-libs/gst/opencv/meson.build
@@ -9,9 +9,9 @@ opencv_headers = [
   'gstopencvvideofilter.h',
 ]
 
-opencv_dep = dependency('opencv', version : '>= 3.0.0', required : get_option('opencv'))
+opencv_dep = dependency('opencv', version : '>= 3.0.0', required : false)
 if not opencv_dep.found()
-  opencv_dep = dependency('opencv4', version : '>= 4.0.0', required : get_option('opencv'))
+  opencv_dep = dependency('opencv4', version : '>= 4.0.0', required : false)
 endif
 if opencv_dep.found()
   gstopencv = library('gstopencv-' + api_version,
@@ -30,4 +30,6 @@ if opencv_dep.found()
     dependencies : [gstvideo_dep, opencv_dep])
 
   install_headers(opencv_headers, subdir : 'gstreamer-1.0/gst/opencv')
+elif get_option('opencv').enabled()
+  error('OpenCV support enabled but required dependencies were not found.')
 endif
-- 
2.21.0

